src/ddmd/expression.d

```diff
@@ -15409,7 +15410,21 @@ extern (C++) final class CmpExp : BinExp
             }
             if ((t1.ty == Tarray || t1.ty == Tsarray) && (t2.ty == Tarray || t2.ty == Tsarray))
             {
-                semanticTypeInfo(sc, t1.nextOf());
+                // Lower to object.__cmp(e1, e2)
+                Expression al = new IdentifierExp(loc, Id.empty);
+                al = new DotIdExp(loc, al, Id.object);
+                al = new DotIdExp(loc, al, Id.__cmp);
+                al = al.semantic(sc);
+
+                auto arguments = new Expressions();
+                arguments.push(e1);
+                arguments.push(e2);
+
+                al = new CallExp(loc, al, arguments);
+                al = new CmpExp(op, loc, al, new IntegerExp(0));
+                al = al.semantic(sc);
```

Comment:
If I comment this line, test ```fail_compilation/ice11755.d``` should pass as the output is the one desired. However, ```druntime``` tests will fail.
 Calling semantic  generates new output which doesn't match the one expected. What is the best approach in this case?